custom/aaa/dbaccessors.py

```diff
@@ -0,0 +1,336 @@
+            'dateOfDischarge': 'N/A',
+            'typeOfDelivery': 'N/A',
+            'timeOfDischarge': 'N/A',
+            'placeOfBirth': 'N/A',
+            'deliveryComplications': 'N/A',
+            'placeOfDelivery': 'N/A',
+            'complicationDetails': 'N/A',
+            'hospitalType': 'N/A',
+        }
+
+    def postnatal_care_details(self):
+        return [{
+            'pncDate': '2019-08-20',
+            'postpartumHeamorrhage': 0,
+            'fever': 1,
+            'convulsions': 0,
+            'abdominalPain': 0,
+            'painfulUrination': 0,
+            'congestedBreasts': 1,
+            'painfulNipples': 0,
+            'otherBreastsIssues': 0,
+            'managingBreastProblems': 0,
+            'increasingFoodIntake': 1,
+            'possibleMaternalComplications': 1,
+            'beneficiaryStartedEating': 0,
+        }]
+
+    def antenatal_care_details(self):
+        return [{
+            'ancDate': 'N/A',
+            'ancLocation': 'N/A',
+            'pwWeight': 'N/A',
+            'bloodPressure': 'N/A',
+            'hb': 'N/A',
+            'abdominalExamination': 'N/A',
+            'abnormalitiesDetected': 'N/A',
+        }]
+
+
+class EligibleCoupleQueryHelper(object):
+    def __init__(self, domain, person_case_id, month_end):
+        self.domain = domain
+        self.person_case_id = person_case_id
+        self.month_end = month_end
+
+    def eligible_couple_details(self):
+        woman = Woman.objects.get(domain=self.domain, person_case_id=self.person_case_id)
+        children_by_sex = Child.objects.filter(mother_case_id=self.person_case_id).aggregate(
+            male=Sum(Case(When(sex='M', then=1), output_field=IntegerField())),
+            female=Sum(Case(When(sex='F', then=1), output_field=IntegerField())),
+        )
+        try:
+            male_children_died = int(woman.num_male_children_died)
+        except TypeError:
+            male_children_died = 0
+        try:
+            female_children_died = int(woman.num_female_children_died)
+        except TypeError:
+            female_children_died = 0
+
+        data = {
+            'maleChildrenBorn': max(0, children_by_sex['male']) + max(0, male_children_died),
+            'femaleChildrenBorn': max(0, children_by_sex['female']) + max(0, female_children_died),
+            'maleChildrenAlive': max(0, children_by_sex['male']),
+            'femaleChildrenAlive': max(0, children_by_sex['female']),
+            'familyPlaningMethod': 'N/A',
+            'familyPlanningMethodDate': 'N/A',
+            'ashaVisit': 'N/A',
+            'previousFamilyPlanningMethod': 'N/A',
+            'preferredFamilyPlaningMethod': 'N/A',
+        }
+
+        history = WomanHistory.objects.filter(person_case_id=self.person_case_id).first()
+        safe_history = {}
+        if history:
+            safe_history = history.date_filter(self.month_end)
+
+        planning_methods = safe_history.get('family_planning_method')
```

Comment:
Some pretty awkward case handling here, but haven't thought of a better way at the moment